packages/launcher/src/renderer/launcher/wallets/WalletAddLedgerModal.js

```diff
@@ -94,65 +142,141 @@ export default class WalletAddLedgerModal extends Component<Props, State> {
+          this.props.onSuccess ? this.props.onSuccess() : this.props.onClose()
         }
       },
       onError: err => {
+        // console.log(err)
         this.displayError(err)
       },
     })
   }
 
   displayError(error: Error | PayloadError) {
     this.setState({
+      saving: false,
       errorMsg: error.message,
     })
   }
 
+  getAddresses = memoize((wallets: Array<Wallet>) => {
+    const addresses = flatten(wallets.map(w => w.accounts))
+    return addresses
+  })
+
+  hasAddress = memoize((addresses: Array<Object>, address: string) => {
+    return addresses.filter(acc => address === acc.address).length > 0
+  })
+
   renderAccounts() {
+    if (!this.state.connected) {
+      return (
+        <ActivityContainer>
+          <Text variant={['center', 'modalText', 'marginBottom20']}>
+            Please unlock your Ledger and open the Ethereum app.
+          </Text>
+          <Loader />
+        </ActivityContainer>
+      )
+    }
+
+    if (this.state.saving) {
+      return (
+        <ActivityContainer>
+          <Text variant={['center', 'modalText', 'marginBottom20']}>
+            Saving your accounts...
+          </Text>
+          <Loader />
+        </ActivityContainer>
+      )
+    }
+
     if (this.state.fetchingAccounts) {
       return (
         <ActivityContainer>
-          <ActivityIndicator />
+          <Text variant={['center', 'modalText', 'marginBottom20']}>
+            Fetching accounts...
+          </Text>
+          <Loader />
         </ActivityContainer>
       )
     }
+
+    const addresses = this.getAddresses(this.props.wallets)
+
     if (this.state.accounts) {
       const accounts = this.state.accounts.map((a, i) => {
-        const onPress = () => this.onSelectAccount(i)
+        const index = i + (this.state.currentPage - 1) * 10
+        const onPress = () => this.onSelectAccount(index)
+        const selected = this.state.selectedItems.indexOf(index) >= 0
+        const disabled = this.hasAddress(addresses, a)
         return (
-          <TouchableOpacity key={a} onPress={onPress}>
-            <AccountItem>{a}</AccountItem>
-          </TouchableOpacity>
+          <AccountItem odd={i % 2} key={a}>
+            <Checkbox
+              variant="mono"
+              defaultValue={selected || disabled}
+              onChange={onPress}
+              label={a}
+              disabled={disabled}
+            />
+          </AccountItem>
         )
       })
-      return <AccountsContainer>{accounts}</AccountsContainer>
+      return (
+        <>
+          <Text variant={['center', 'modalText', 'marginBottom20']}>
+            Select one or multiple addresses
+          </Text>
+          <AccountsContainer>
+            <ScrollView>{accounts}</ScrollView>
+          </AccountsContainer>
+          <Pagination
+            numPages={1000}
+            maxDisplay={10}
+            defaultPage={this.state.currentPage}
+            label="Page"
+            // $FlowFixMe
```

Comment:
What is the flow issue here?

Reply:
I just wrapped it in another function... should be fine now